applications/minotari_app_grpc/src/authentication/basic_auth.rs

```diff
@@ -36,8 +41,17 @@ pub struct BasicAuthCredentials {
 }
 
 impl BasicAuthCredentials {
-    pub fn new(user_name: String, password: SafePassword) -> Self {
-        Self { user_name, password }
+    pub fn new(user_name: String, password: SafePassword) -> Result<Self, BasicAuthError> {
```

Comment:
It might be helpful to rename `password` to something that reflects its status as a PHC string, not the "raw" password. This was a bit confusing when I first read this code.